README.rst

```diff
@@ -9,12 +9,91 @@ mily
         :target: https://pypi.python.org/pypi/mily
 
 
-A collection of Qt building blocks for scientific GUIs.
+A collection of Qt building blocks for scientific and control GUIs in Python
 
 * Free software: 3-clause BSD license
 * Documentation: (COMING SOON!) https://NSLS-II.github.io/mily.
 
-Features
---------
+Project Goals
+-------------
 
-* TODO
+Across the light sources we have a range of Qt based tools covering
+the full range of application: from engineering screens, to analysis
+GUIs.  Each of these projects wrapping and extending the base Qt
+Widgets in very similar ways that are both a duplication of effort and
+may reduce the ease with which we can combine the projects.  The
+high-level goals of this project are:
+
+ - code re-use between projects
+ - similar look-and-feel across fleet
+ - ability to make applications with cross-cutting functionality
+ - make sure we are all talking to each other
+
+
+
+In Scope
+++++++++
+
+- "smarter" versions of core Qt widgets (ex, float-aware ``QSlider``,
+  spin-box + label, display alarm state)
+- domain-specific visualization
+- helper functions for managing Widgets / windows
+- dialog to show an exception to the user
+- auto-form generation tools
+
+Out of Scope
+++++++++++++
+
+- data reduction, transformation, or analysis code
+- control-system aware code
+- data I/O code
+- business logic
+
+Related projects
```

Comment:
Could be related: https://github.com/leofang/ptycho_gui, A GUI based on PyQt5 to provide control and visualization of ptychography reconstruction. 

Below is a short description we wrote for the GUI (unpublished yet, sorry):
> For users’ convenience, we also provide a GUI for pre-processing the scan data, executing an ptychographic reconstruction, monitoring the result as the reconstruction progresses, and post-processing the obtained probe and object, as shown in Fig. 6. The overall framework of the GUI is built with PyQt5, and previewing the scan images, plotting error metrics, visualizing resultant images and so on, are handled using Matplotlib.
>
> In the main panel, see Fig. 6a, users can load the scan data to be reconstructed (either from the NSLS-II Databroker [30] or an existing HDF5 file) and check the associated experimental parameters. Users can open the pre-processing panel, see Fig. 6b, with the loaded data, and it provides both automatic/manual bad-pixel and region-of-interest detection on the raw diffraction data. The pre- processing panel is also equipped with convenient functionalities such as panning and zooming in/out for better inspection of the images. After that, in the main panel, users can select up to two ptychography algorithms and configure the reconstruction parameters, such as the floating- point precision, the number of iterations, the number of GPUs (or CPUs) the code is to run on, etc. Once the reconstruction starts, the monitor panel in Fig. 6c is popped up so that users can monitor/navigate the progress of the reconstruction in a streaming (or static) manner. The communication between the frontend (GUI) and the backend (CPU/GPU ptychography code) is done via POSIX shared memory, enabled by the Python package posix_ipc which provides a wrapper over the POSIX API.
>
> The clean separation of UI design, logic implementation, and actual computation allows the GUI to be easily maintained. To smoothly manage all functionalities provided in the GUI, it is multi- threaded and managed using the signal-slot and the QThread mechanisms in the Qt framework, even though the GUI itself only runs on one CPU core. Lastly, the GUI also provides a batch mode, with which users can process over several sets of scan data in the same setting.

ps. sorry for late reply